lib/measured/unit_system_builder.rb

```diff
@@ -8,12 +8,53 @@ def unit(unit_name, aliases: [], value: nil)
     nil
   end
 
+  def si_unit(unit_name, aliases: [], value: nil)
+    @units += build_si_units(unit_name, aliases: aliases, value: value)
+    nil
+  end
+
   def build
     Measured::UnitSystem.new(@units)
   end
 
   private
 
+  def self.prefixes
+    [
+      ['y', 'yocto', -24],
+      ['z', 'zepto', -21],
+      ['a', 'atto', -18],
+      ['f', 'femto', -15],
+      ['p', 'pico', -12],
+      ['n', 'nano', -9],
+      ['μ', 'micro', -6],
+      ['m', 'milli', -3],
+      ['c', 'centi', -2],
+      ['d', 'deci', -1],
+      ['da', 'deca', 1],
+      ['h', 'hecto', 2],
+      ['k', 'kilo', 3],
+      ['M', 'mega', 6],
+      ['G', 'giga', 9],
+      ['T', 'tera', 12],
+      ['P', 'peta', 15],
+      ['E', 'exa', 18],
+      ['Z', 'zetta', 21],
+      ['Y', 'yotta', 24]
+    ]
+  end
+
+  def build_si_units(name, aliases: [], value: nil)
+    size = value.split(' ')[0] unless !value
+    size ||= 1
+    si_units = [build_unit(name, aliases: aliases, value: value)]
```

Comment:
Right. In my solution, the "identity" element of the SI-prefixed units (in your example, `long_ton`) would be built with the given value:

```ruby
si_units = [build_unit(name, aliases: aliases, value: value)]
```

In your example, the given `value` is `"2240 lb"`, and the unit class will properly deal with parsing that string. Every other unit though can be given conversion values based on `long_ton`:

```ruby
si_units << build_unit("#{short}#{name}", aliases: long_names, value: "10e#{exp} #{name}")
```

For example, for `mega` the above line would evaluate to:

```ruby
si_units << build_unit("mlong_ton", aliases: [], value: "10e6 long_ton")
```